### PR TITLE
docs: update /contributing docs

### PR DESCRIPTION
**Pull request by meganesu**
## Description

This PR updates the docs under the /contributing part of the Gatsby site, to align with recent changes from the docs reorg project.

More specifically, this PR:

* Reorganizes files into subfolders, to make them easier to find.
* Updates the docs templates to match the new doc types: How-To Guides, Reference Guides, Conceptual Guides, and the Tutorial.

TODO:

- [ ] Add redirects for files that were moved or deleted.

### CODE REVIEW

File: docs/contributing/docs-contributions/how-to-write-a-plugin-readme.md [hunk 1]
```diff
@@ -3,6 +3,8 @@ title: How to Write a Plugin README
 issue: https://github.com/gatsbyjs/gatsby/issues/21599
```
Comment (LekoArts) [inline docs/contributing/docs-contributions/how-to-write-a-plugin-readme.md:3]: Is this still needed?

File: docs/contributing/docs-contributions/how-to-write-a-reference-guide.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+In 2-3 sentences, give a high-level description of what this feature does,
+why it's important, and when it might be helpful for users.
+
+## Feature Name
+
+Break up this section into multiple headings, as needed.
+
+Here are some general tips for helpful things to include:
+
+- Diagrams or other visuals, to show key processes or architectures.
```
Comment (LekoArts) [inline docs/contributing/docs-contributions/how-to-write-a-reference-guide.md:48]: Maybe call out that those diagrams should have a great alt text?

File: docs/contributing/stub-list.md [hunk 3]
```diff
@@ -1,10 +0,0 @@
----
```
Comment (LekoArts) [inline docs/contributing/stub-list.md:1]: If that was deleted, were all stubs references removed from the docs?

File: docs/docs/how-to/plugins-and-themes/using-a-plugin-in-your-site.md [hunk 4]
```diff
@@ -1,56 +1,93 @@
 ---
-title: Using a Plugin in Your Site
+title: Add a Plugin to Your Site
```
Comment (LekoArts) [inline docs/docs/how-to/plugins-and-themes/using-a-plugin-in-your-site.md:2]: Should this doc change be part of a separate PR?

**Review by LekoArts — CHANGES_REQUESTED** [review]
Some smaller comments but overall great improvements!

File: docs/contributing/docs-contributions/docs-structure.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+- [How to Write a Reference Guide](/contributing/docs-contributions/how-to-write-a-reference-guide/)
+- [How to Write a Conceptual Guide](/contributing/docs-contributions/how-to-write-a-conceptual-guide/)
+- [How to Write a Tutorial](/contributing/docs-contributions/how-to-write-a-tutorial)
+
+The guides above provide more details about each type of documentation. They also include templates and tips for how to write them. Use these resources to make sure your docs contribution fits the overall documentation structure.
+
+Sometimes, a single concept will require some combination of these documentation types. For example, the `gatsby-plugin-image` documentation includes the following pieces:
+
+- A Reference Guide to explain the plugin API and all its options.
+- A How-To Guide to show practical examples of how to use the plugin.
```
[the remark this answers is not in the document]
Comment (meganesu) [reply docs/contributing/docs-contributions/docs-structure.md:36]: Great idea!

```suggestion
- [Gatsby Image plugin](/docs/reference/built-in-components/gatsby-plugin-image/): A Reference Guide to explain the plugin API and all its options.
- [Using the Gatsby Image plugin](/docs/how-to/images-and-media/using-gatsby-plugin-image/): A How-To Guide to show practical examples of how to use the plugin.
```

File: docs/contributing/docs-contributions/how-to-write-a-conceptual-guide.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
+---
+title: How to Write a Conceptual Guide
+---
+
+## What is a Conceptual Guide?
+
+A Conceptual Guide provides a big-picture explanation of higher-level Gatsby concepts. Conceptual Guides are most useful for readers who want to build a deeper understanding of a particular topic.
```
[the remark this answers is not in the document]
Comment (meganesu) [reply docs/contributing/docs-contributions/how-to-write-a-conceptual-guide.md:7]: I've been capitalizing them everywhere, like a proper noun. I was trying to show that it's the name of a specific thing, not just a general term. (#branding)

Does it feel off?

File: docs/contributing/docs-contributions/how-to-write-a-how-to-guide.md [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+
+## How-To Guide template
+
+The title of your How-To Guide should specify what it helps readers do. You should be able to add the words "how to" to the beginning and have it make sense. For example, if your guide is about "how to create a new layout component", the title would be "Create a New Layout Component."
+
+A How-To Guide should include the following sections:
+
+- **Introduction:** Provide some quick background information to help readers understand 1) what this How-To Guide will help them achieve and 2) why that's useful.
+- **Prerequisites:** Any additional information or setup readers will have to do before they can make use of this How-To Guide.
+- **Directions:**
```
[the remark this answers is not in the document]
Comment (meganesu) [reply docs/contributing/docs-contributions/how-to-write-a-how-to-guide.md:27]: Oops! Thanks 😅

```suggestion
- **Directions:** List out the steps that readers to follow to complete the task. Use code blocks to show exactly what readers should type in their own projects.
```

File: docs/contributing/docs-contributions/how-to-write-a-how-to-guide.md [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+plugin, or already know a certain skill? List those things here.
+
+## Directions
+
+This section should be a step-by-step list of instructions for how readers
+can achieve the goal.
+
+1. Do the first thing.
+2. Do the next thing.
+
```
Comment (meganesu) [inline docs/contributing/docs-contributions/how-to-write-a-how-to-guide.md:61]:
```suggestion

1. Do the first thing.
2. Do the next thing.

Use code blocks to show exactly what readers should type. The
Gatsby Style Guide includes tips for best practices when using code
blocks:
https://www.gatsbyjs.com/contributing/gatsby-style-guide/#format-code-blocks-inline-code-and-images

```

File: docs/contributing/docs-contributions/how-to-write-a-how-to-guide.md [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+## Prerequisites
+
+If applicable, list any prerequisites to reading and understanding your
+article.
+
+Does the reader need to read another document first, install a particular
+plugin, or already know a certain skill? List those things here.
+
+## Directions
```
[the remark this answers is not in the document]
Comment (meganesu) [reply docs/contributing/docs-contributions/how-to-write-a-how-to-guide.md:54]: Good idea. Added a note below

File: docs/contributing/docs-contributions/how-to-write-a-how-to-guide.md [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+
+1. The first step of the first thing.
+1. The second step of the first thing.
+
+### Step 2: Do some other thing.
+
+1. The first step of some other thing.
+1. The second step of some other thing.
+
+## Additional Resources
```
[the remark this answers is not in the document]
Comment (meganesu) [reply docs/contributing/docs-contributions/how-to-write-a-how-to-guide.md:75]: I don't have a super strong stance on this either way. My first thought is that we might be able to handle this on a case-by-case basis. Like if they have a really good article on their own blog, I don't mind if they include it.

Do you have a preference?

File: docs/contributing/docs-contributions/how-to-write-a-plugin-readme.md (hunk 1 above)
Comment (meganesu) [reply docs/contributing/docs-contributions/how-to-write-a-plugin-readme.md:3]: Honestly, I left it because I'm not sure what this frontmatter does, and was mainly focused on content this time through 😅

Comment (meganesu) [reply docs/contributing/docs-contributions/how-to-write-a-plugin-readme.md:3]: Do you know what it's for? I don't mind deleting it!

File: docs/contributing/docs-contributions/how-to-write-a-reference-guide.md (hunk 2 above)
Comment (meganesu) [reply docs/contributing/docs-contributions/how-to-write-a-reference-guide.md:48]:
```suggestion
- Diagrams or other visuals, to show key processes or architectures.
  - Make sure to include alt text for accessibility! For
    help writing great alt text, refer to the W3C alt decision
    tree: https://www.w3.org/WAI/tutorials/images/decision-tree/
```

File: docs/docs/how-to/plugins-and-themes/using-a-plugin-in-your-site.md (hunk 4 above)
Comment (meganesu) [reply docs/docs/how-to/plugins-and-themes/using-a-plugin-in-your-site.md:2]: I think it's okay to leave in here, since I updated it as part of the "how to write a how-to guide" contributing doc. To make something that fits the template.

File: docs/contributing/blog-contributions.md [hunk 11]
```diff
@@ -2,7 +2,9 @@
 title: Blog Contributions
 ---
 
-If you'd like to contribute a post to the Gatsby blog, please review the process and guidelines outlined below and submit your idea for the post to our [Gatsby blog proposal form](https://airtable.com/shr3449954866i3iF)
+Blog posts are a great way to share your own experiences working with Gatsby at work or on a personal project.
+
+If you'd like to contribute a post to the Gatsby blog, please review the process and guidelines outlined below and submit your post idea using the [Gatsby blog proposal form](https://airtable.com/shr3449954866i3iF).
```
[the remark this answers is not in the document]
Comment (meganesu) [reply docs/contributing/blog-contributions.md:7]: I checked in with Michelle back in December and she said it still looked right. I'll ask her again just to be sure.

File: docs/contributing/stub-list.md (hunk 3 above)
Comment (meganesu) [reply docs/contributing/stub-list.md:1]: Good point. I just deleted and set up redirects for all the ones except the ones under "using gatsby professionally". (There's a different story to figure out what to do with those ones, since they should probably be pulled out of the docs site.)

File: docs/contributing/docs-contributions/how-to-write-a-conceptual-guide.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (LekoArts) [reply docs/contributing/docs-contributions/how-to-write-a-conceptual-guide.md:7]: It not necessarily felt "off" but seemed unusual as I'm only used to see this with "actual" brand names/proper nouns. But I don't have a strong opinion on this, feel free to keep it this way.

File: docs/contributing/docs-contributions/how-to-write-a-how-to-guide.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (LekoArts) [reply docs/contributing/docs-contributions/how-to-write-a-how-to-guide.md:80]: Yeah, it'll be a case-by-case basis anyways. Maybe rephrase it then to mention that content _in the best case_ is evergreen (not quickly outdated), high quality, following our own CoC (strong language, a11y)

File: docs/contributing/docs-contributions/how-to-write-a-plugin-readme.md (hunk 1 above)
Comment (LekoArts) [reply docs/contributing/docs-contributions/how-to-write-a-plugin-readme.md:3]: afaik it's for displaying a note about the stub issue. See at the bottom here: https://www.gatsbyjs.com/contributing/how-to-write-a-plugin-readme/

> See the issue relating to this stub on GitHub

So if we deleted all stubs then we can get rid of these `issue` in frontmatter

File: docs/docs/how-to/plugins-and-themes/using-a-plugin-in-your-site.md (hunk 4 above)
Comment (LekoArts) [reply docs/docs/how-to/plugins-and-themes/using-a-plugin-in-your-site.md:2]: Okay 👍

**Comment by LekoArts** [comment]
Now that the stubs are deleted, I think some issues here are/were the respective stub issues: https://github.com/gatsbyjs/gatsby/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22type%3A+documentation%22

File: docs/contributing/blog-contributions.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (meganesu) [reply docs/contributing/blog-contributions.md:7]: Sounds like she and Sam are revisiting the blog-specific process. I think it makes sense to add those updates in a future PR, if there are any.

File: docs/contributing/docs-contributions/how-to-write-a-plugin-readme.md (hunk 1 above)
Comment (meganesu) [reply docs/contributing/docs-contributions/how-to-write-a-plugin-readme.md:3]: Gotcha, thanks! I'll delete it ✂️

Comment (meganesu) [reply docs/contributing/docs-contributions/how-to-write-a-plugin-readme.md]: (Hm that didn't work the way I thought it would. That's what I get for using the UI.)

File: docs/contributing/docs-contributions/how-to-write-a-how-to-guide.md [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+
+Include other resources you think readers would benefit from or next steps
+they might want to take after reading your How-To Guide. You can also
+mention any resources that helped you write the article (blog posts, outside
+tutorials, etc.).
+
```
Comment (meganesu) [inline docs/contributing/docs-contributions/how-to-write-a-how-to-guide.md:94]:
```suggestion
Include other resources you think readers would benefit from or next steps
they might want to take after reading your How-To Guide. You can also
mention any resources that helped you write the article (blog posts, outside
tutorials, etc.).

Best-case scenario, these should be high-quality, evergreen
(not quickly outdated) resources.
```

File: docs/contributing/docs-contributions/how-to-write-a-how-to-guide.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (meganesu) [reply docs/contributing/docs-contributions/how-to-write-a-how-to-guide.md:88]: Can do! Added a line about that. (I left out the part about it being in English. We can add it in later if it ends up being a problem!)

**Comment by meganesu** [comment]
> Now that the stubs are deleted, I think some issues here are/were the respective stub issues: https://github.com/gatsbyjs/gatsby/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22type%3A+documentation%22

@LekoArts Good catch. I closed the issues.

I think I've addressed all the comments. Thanks for the review! Could you approve this? I'll handle coordinating merging this and the mansion PR.
